examples_tests/0.Polyphase/main.cpp

```diff
@@ -0,0 +1,144 @@
+{
+	IImage::SCreationParams imageParams = {};
+	imageParams.flags = static_cast<IImage::E_CREATE_FLAGS>(0u);
+	imageParams.type = IImage::ET_2D;
+	imageParams.format = asset::EF_R32_SFLOAT;
+	imageParams.extent = { dims[0], dims[1], 1 };
+	imageParams.mipLevels = 1u;
+	imageParams.arrayLayers = 1u;
+	imageParams.samples = asset::ICPUImage::ESCF_1_BIT;
+
+	auto imageRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<asset::IImage::SBufferCopy>>(1ull);
+	auto& region = (*imageRegions)[0];
+	region.bufferImageHeight = 0u;
+	region.bufferOffset = 0ull;
+	region.bufferRowLength = dims[0];
+	region.imageExtent = { dims[0], dims[1], 1u };
+	region.imageOffset = { 0u, 0u, 0u };
+	region.imageSubresource.baseArrayLayer = 0u;
+	region.imageSubresource.layerCount = 1u;
+	region.imageSubresource.mipLevel = 0;
+
+	size_t bufferSize = asset::getTexelOrBlockBytesize(imageParams.format) * region.imageExtent.width * region.imageExtent.height;
+	auto imageBuffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(bufferSize);
+	core::smart_refctd_ptr<ICPUImage> image = ICPUImage::create(std::move(imageParams));
+	image->setBufferAndRegions(core::smart_refctd_ptr(imageBuffer), imageRegions);
+
+	return image;
+}
+
+void blit(core::smart_refctd_ptr<ICPUImage> inImage, core::smart_refctd_ptr<ICPUImage> outImage, bool useLUT)
+{
+	const core::vectorSIMDf scaleX(3.f, 1.f, 1.f, 1.f);
+	const core::vectorSIMDf scaleY(1.f, 3.f, 1.f, 1.f);
+	const core::vectorSIMDf scaleZ(1.f, 1.f, 1.f, 1.f);
+
+	auto kernelX = ScaledBoxKernel(scaleX, CBoxImageFilterKernel()); // [-3/2, 3/2]
+	auto kernelY = ScaledBoxKernel(scaleY, CBoxImageFilterKernel()); // [-3/2, 3/2]
+	auto kernelZ = ScaledBoxKernel(scaleZ, CBoxImageFilterKernel()); // [-1/2, 1/2]
+
+	BlitFilter::state_type blitFilterState(std::move(kernelX), std::move(kernelY), std::move(kernelZ));
+	blitFilterState.inOffsetBaseLayer = core::vectorSIMDu32();
+	blitFilterState.inExtentLayerCount = core::vectorSIMDu32(0u, 0u, 0u, inImage->getCreationParameters().arrayLayers) + inImage->getMipSize();
+	blitFilterState.inImage = inImage.get();
+
+	blitFilterState.outOffsetBaseLayer = core::vectorSIMDu32();
+	blitFilterState.outExtentLayerCount = core::vectorSIMDu32(0u, 0u, 0u, outImage->getCreationParameters().arrayLayers) + outImage->getMipSize();
+	blitFilterState.outImage = outImage.get();
+
+	blitFilterState.axisWraps[0] = ISampler::ETC_CLAMP_TO_EDGE;
+	blitFilterState.axisWraps[1] = ISampler::ETC_CLAMP_TO_EDGE;
+	blitFilterState.axisWraps[2] = ISampler::ETC_CLAMP_TO_EDGE;
+	blitFilterState.borderColor = ISampler::E_TEXTURE_BORDER_COLOR::ETBC_FLOAT_OPAQUE_WHITE;
+
+	blitFilterState.enableLUTUsage = useLUT;
+
+	blitFilterState.scratchMemoryByteSize = BlitFilter::getRequiredScratchByteSize(&blitFilterState);
+	blitFilterState.scratchMemory = reinterpret_cast<uint8_t*>(_NBL_ALIGNED_MALLOC(blitFilterState.scratchMemoryByteSize, 32));
```

Comment:
so here you'd want the LUT to be counted into the scratch size and taken from an offset

And yes the LUT population function would need to be split off from `execute` so the LUT could be reused for blitting images with identical phases (usually this means identical source to destination extent rations), obviously not by the filter, the user would have invent such a caching scheme.

Reply:
If I understand this correctly, I think it is resolved now.